obs2aptly/src/lib.rs

```diff
@@ -612,6 +644,104 @@ impl SyncActions {
     pub fn actions(&self) -> &[SyncAction] {
         &self.actions
     }
+
+    #[tracing::instrument(skip_all)]
+    async fn upload_file(&self, directory: String, path: &Path) -> Result<()> {
+        info!("Uploading {}", path.display());
+
+        let filename = path
+            .file_name()
+            .and_then(|f| f.to_str())
+            .map(|f| f.to_owned())
+            .ok_or_else(|| eyre!("Invalid path"))?;
+
+        let file = File::open(path).await?;
+        let body = reqwest::Body::wrap_stream(FramedRead::new(file, BytesCodec::new()));
```

Comment:
probably the aptly-rest api should have a helper for this; the obs2aptly tooling really shouldn't have to know reqwest is used underneath :)